extensions/src/main/java/org/pentaho/platform/osgi/KarafBoot.java

```diff
@@ -224,24 +239,22 @@
         root = destDir.toURI().getPath();
       }
 
-      configureSystemProperties( solutionRootPath, root );
-
-      String customLocation = root + "/etc/custom.properties";
-      expandSystemPackages( customLocation );
+      this.karafCustomProperties = getCustomProperties( root );
```

Comment:
Perhaps call `setCustomProperties` to set this.
Also, `getCustomProperties` should read `this.karafCustomProperties` alone, and a new method called something like `readCustomProperties` would read the properties.